tests/Integration/Behaviour/Features/Context/Domain/CategoryFeatureContext.php

```diff
@@ -381,4 +545,103 @@ private function mapDataToEditableCategory(
             $subcategories
         );
     }
+
+    /**
+     * @param array $testCaseData
+     *
+     * @return int
+     */
+    private function getParentCategoryId(array $testCaseData)
+    {
+        $parentCategoryId = null;
+        if (isset($testCaseData['Parent category'])) {
+            /** @var CategoryTreeChoiceProvider $categoryTreeChoiceProvider */
+            $categoryTreeChoiceProvider = $this->container->get(
+                'prestashop.adapter.form.choice_provider.category_tree_choice_provider');
+            $categoryTreeIterator = new CategoryTreeIterator($categoryTreeChoiceProvider);
+            $parentCategoryId = $categoryTreeIterator->getCategoryId($testCaseData['Parent category']);
+        }
+        if ($parentCategoryId === null) {
+            $parentCategoryId = CategoryTreeIterator::ROOT_CATEGORY_ID;
+        }
+
+        return $parentCategoryId;
+    }
+
+    /**
+     * @param array $testCaseData
+     *
+     * @return array
+     */
+    private function getGroupAssociationIds(array $testCaseData): array
+    {
+        /** @var GroupByIdChoiceProvider $groupByIdChoiceProvider */
+        $groupByIdChoiceProvider = $this->container->get(
+            'prestashop.adapter.form.choice_provider.group_by_id_choice_provider'
+        );
+        $groupChoicesArray = $groupByIdChoiceProvider->getChoices();
+
+        $groupAssociationIds = [];
+        if (isset($testCaseData['Group access'])) {
+            $groupAssociations = explode(',', $testCaseData['Group access']);
+            foreach ($groupAssociations as $groupAssociation) {
+                $groupAssociationIds[] = (int) $groupChoicesArray[$groupAssociation];
+            }
+        } else {
+            $groupAssociationIds = array(
+                0 => '1',
+                1 => '2',
+                2 => '3',
+            );
+        }
+
+        return $groupAssociationIds;
+    }
+
+    /**
+     * @param array $testCaseData
+     * @param int $categoryId
+     */
+    private function pretendImageUploading(array $testCaseData, int $categoryId): void
+    {
+        $categoryCoverImage = $testCaseData['Category cover image'];
+        // could not use handler because it uses move_uploaded_file which allows only POST upload
+        /** @var Kernel $kernel */
+        $kernel = $this->getContainer()->get('kernel');
+        copy(
+            $kernel->getRootDir() . '/../img/' . $categoryCoverImage,
+            _PS_CAT_IMG_DIR_ . $categoryId . '.jpg'
+        );
+        copy(
+            $kernel->getRootDir() . '/../img/' . $categoryCoverImage,
+            _PS_CAT_IMG_DIR_ . $categoryId . '-' . stripslashes($categoryCoverImage) . '.jpg'
+        );
```

Comment:
I dont think its a good idea to actually move images in the test scenarios - the best option would be to mock and imitate actual image moving from source to destination. If its not possible when better not test this case at all

Reply:
Thanks Tomas, for now commented out scenarios using this method

Reply:
https://github.com/symfony/symfony/blob/8e9aafc84fc741d966607bff5b18fdaaf0ed2987/src/Symfony/Component/HttpFoundation/File/UploadedFile.php#L19-L53

Reply:
@jolelievre, @PierreRambaud - currently our legacy classes forces to upload the file via HTTP form. That's why cannot use UploadedFile in test mode. Any creative ideas to overcome this issue would be helpful. 

Also @matks suggested a solution below.

Reply:
Because we are not using `\copy` we can mock with namespace, but it's quite ugly.
I'm more in favor of refactoring like he said: https://github.com/PrestaShop/PrestaShop/pull/17018#issuecomment-571985497